StitchCore/Tests/StitchCoreTests/Auth/Internal/MockCoreStitchAuth.swift

```diff
@@ -0,0 +1,64 @@
+import Foundation
+import MongoSwift
+import MockUtils
+@testable import StitchCore
+
+struct StubAuthRoutes: StitchAuthRoutes {
+    var sessionRoute: String = ""
+    
+    var profileRoute: String = ""
+    
+    var baseAuthRoute: String = ""
+    
+    func authProviderRoute(withProviderName providerName: String) -> String {
+        return ""
+    }
+    
+    func authProviderLoginRoute(withProviderName providerName: String) -> String {
+        return ""
+    }
+    
+    func authProviderLinkRoute(withProviderName providerName: String) -> String {
+        return ""
+    }
+}
+
+final class StubStitchRequestClient: StitchRequestClient {
+    init() { }
+    
+    init(baseURL: String, transport: Transport, defaultRequestTimeout: TimeInterval) { }
+    
+    func doRequest(_ stitchReq: StitchRequest) throws -> Response {
+        return Response.init(statusCode: 500, headers: [:], body: nil)
+    }
+}
+
+final class MockCoreStitchAuth<TStitchUser>: CoreStitchAuth<TStitchUser> where TStitchUser: CoreStitchUser {
+    // concrete classes in Swift are a little tricky to mock. we can't do a true mock, since super.init must always be
+    // called in Swift. This init makes sure that the super init runs without an error.
+    public init() {
+        self.getAuthInfoMock.doReturn(result: nil) // necessary for init() to run without failing
+        try! super.init(requestClient: StubStitchRequestClient.init(),
+                        authRoutes: StubAuthRoutes.init(),
+                        storage: MemoryStorage(),
+                        startRefresherThread: false)
+        self.getAuthInfoMock.clearStubs()
+        self.getAuthInfoMock.clearInvocations()
+    }
+    
+    public var isLoggedInMock = FunctionMockUnit<Bool>()
+    override var isLoggedIn: Bool {
+        return isLoggedInMock.run()
```

Comment:
[q] These artifacts seem to defeat the purpose of mocking a bit. Is there any way we can avoid this much dependency injection?

Reply:
I don't think it completely defeats the purpose. We're wrapping a lot of functionality in the FunctionMockUnit, and IMO a little bit of test boilerplate is better than a lot of test boilerplate. This also seems to be the approach of other mocking libraries in Swift (including Cuckoo), it's just that those mocking libraries do code generation to generate the artifacts. In the long term we could look into code generation but I think this is fine for now.